File: generator_fib.js
let fibGeneratorInstance = function* fibGenerator() {
	let a = 0; let b = 1;
	yield a; yield b;
	while ( b != 'Infinity' ) {
		const tmp = b;
		b = a + b;
		a = tmp;
		yield b;
	}
}();
for ( const fib of fibGeneratorInstance ) {
	console.log( fib );
}

File: generator_fib.cpp
#include <generator>
#include <coroutine>
#include <iostream>
#include <ranges>
#include <cstdint>

// <generator> header isn't available anywhere at the time of writing, so this is only a guess rn
std::generator<u_int64_t> fib_generator() {
	u_int64_t a = 0;
	u_int64_t b = 1;
	co_yield a; co_yield b;
	while (true) {
		const u_int64_t tmp = b;
		b = a + b;
		a = tmp;
		co_yield b;
	}
}

auto main() -> int {
    for (const char ch : fib_generator() | std::views::take(26))
        std::cout << ch << ' ';
    std::cout << '\n';
}